core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/ldap/LDAPConnectionContext.java

```diff
@@ -211,14 +211,20 @@ public LDAPConnectionContext(RealmConfiguration realmConfig) throws UserStoreExc
 
         // Set socket factory for not verifying SSL certificate.
         boolean isSSLCertificateValidationEnabled = true;
-        String valueForSSLCertificateValidationEnabled = realmConfig.getUserStoreProperty(UserStoreConfigConstants.
-                SSLCertificateValidationEnabled);
-        if (StringUtils.isNotEmpty(valueForSSLCertificateValidationEnabled)) {
-            isSSLCertificateValidationEnabled = Boolean.parseBoolean(valueForSSLCertificateValidationEnabled);
+        boolean isSecuredLDAP = false;
+        if (connectionURL != null) {
+            isSecuredLDAP = connectionURL.startsWith("ldaps:");
```

Comment:
Can we use regex, so that this is resilient on CASE.

Reply:
fixed